core-bundle/src/DependencyInjection/Compiler/ConfigureFilesystemPass.php

```diff
@@ -55,7 +55,7 @@ private function mountAdaptersForSymlinks(ContainerBuilder $container, Filesyste
         $projectDir = $parameterBag->resolveValue($parameterBag->get('kernel.project_dir'));
         $uploadDir = $parameterBag->resolveValue($parameterBag->get('contao.upload_path'));
 
-        $finder = (new Finder())->in($projectDir)->directories()->path('/^'.preg_quote($uploadDir, '/').'/');
+        $finder = (new Finder())->in($projectDir)->followLinks()->directories()->path('/^'.preg_quote($uploadDir, '/').'/');
```

Comment:
But wouldn’t this find symlinks inside other symlinks and the code below would break then?
What was the issue with the first solution?

Reply:
You are right, I incorrectly thought that following symlinks was the issue, whereas the problem is actually that the `/files/` folder is a symlink that is being resolved. I am still working on this pull request, will keep you updated.

Reply:
I have just pushed the new commit and a unit test, please have a look.